<launch>
    <param name="use_sim_time" value="true" />


     <param name="robot_description" command="$(find xacro)/xacro $(find wolfgang_description)/urdf/wolfgang.urdf.xacro --inorder" />
    <param name="urdf" value="$(find wolfgang_description)/urdf/wolfgang.urdf" />
    <param name="robot_description_semantic" textfile="$(find wolfgang_moveit_config)/config/wolfgang.srdf" />
    <param name="robot_type_name" value="Wolfgang"/>
    <!--include file="$(find humanoid_league_transform)/launch/transformer.launch" /-->
    <node pkg="rosbag" type="play" name="player" output="screen" args="--topic /tf_static /cmd_vel /joint_states /image_raw /line_relative --clock /home/judith/Dokumente/BitBots/rosbags/2019-01-06-16-55-47.bag" />
    <!--no-walking.bag-->


    <!--node name="rosbaglive" pkg="rosbaglive" type="play.py" args="/home/judith/Dokumente/BitBots/rosbags/init.bag" /-->
    <!--<node name="rosbaglive" pkg="rosbaglive" type="play.py" args="/home/judith/Dokumente/BitBots/rosbags/mitTF/2018-05-09-11-59-29.bag" />-->

    <!--node pkg="tf" type="static_transform_publisher" name="odom_broadcaster" args="42 1 -8 0 0 1 0 odom base_link 100" /-->

    <!-- fake pointcloud -->
    <node name="fake_pointcloud" pkg="bitbots_localization" type="fake_pointcloud.py" />

    <!-- fake laserscan -->
    <include file="$(find bitbots_localization)/launch/pointcloud_to_laserscan.launch" />

    <!-- load map -->
    <node name="map_server" pkg="map_server" type="map_server" output="screen" args="$(find bitbots_localization)/models/d125.yaml"/>

    <!-- particle filter -->
    <node name="amcl_particlefilter"  pkg="amcl" type="amcl" output="screen">
       <param name="min_particles" value="1000"/>
        <param name="max_particles" value="5000"/>
        <param name="kld_err" value="0.0001"/>
        <param name="kld_z" value="0.99"/>

        <param name="update_min_d" value="0.2"/>
        <!--param name="update_min_a" value="0.01"/-->

        <param name="resample_interval" value="2"/>

       <param name="initial_pose_x" value="2.0"/>
        <param name="initial_pose_y" value="3.0"/>
        <param name="initial_pose_a" value="4.7"/>
        <param name="initial_cov_xx" value="4.0"/>
        <param name="initial_cov_yy" value="5.0"/>
        <param name="initial_cov_aa" value="3.14"/>

        <param name="gui_publish_rate" value="1.0"/>

        <param name="use_map_topic" value="true" />
        <param name="first_map_only" value="true"/>

        <!--param name="recovery_alpha_slow" value="0.001"/-->
        <!--param name="recovery_alpha_fast" value="0.1"/-->

        <param name="laser_max_beams" value="100"/>
        <param name="laser_model_type" value="likelihood_field_prob"/>

        <param name="odom_model_type" value="omni-corrected" />
        <param name="odom_alpha1" value="0.9"/>
         <param name="odom_alpha2" value="0.9"/>
        <param name="odom_alpha3" value="0.9"/>
        <param name="odom_alpha4" value="0.9"/>
        <param name="odom_alpha5" value="0.9"/>


        <param name="tf_broadcast" value="true"/>
        <param name="base_frame_id" value="base_footprint" />
    </node>

    <!-- hotfix nomotion update -->
    <node name="amcl_nomotion_update" pkg="bitbots_localization" type="amcl_nomotion_update.py" />

</launch>
